api/include/opentelemetry/trace/span.h

```diff
@@ -1,38 +1,112 @@
-  // time point relative to other spans collected across a distributed system.
+  // Timestamps from the steady clock can be used to most accurately measure a Span's
+  // duration, while timestamps from the system clock can be used to most accurately place a Span's
+  // time point relative to other Spans collected across a distributed system.
   core::SystemTimestamp start_system_time;
   core::SteadyTimestamp start_steady_time;
+
+  // TODO:
+  // Span(Context?) parent;
+  // SpanContext remote_parent;
+  // Links
+  // Attributes
+  SpanKind kind = SpanKind::kInternal;
 };
 
 class Tracer;
 
 /**
- * A span represents a single operation within a trace.
+ * A Span represents a single operation within a Trace.
+ *
+ * Spans *must* be ended by calling End(). Merely deleting a Span does not end
+ * it.
  */
 class Span
 {
 public:
+  // Note that Spans should be created using the Tracer class. Please refer to
+  // tracer.h for documentation.
+  Span()          = default;
   virtual ~Span() = default;
 
+  // Not copiable or movable.
+  Span(const Span &) = delete;
+  Span(Span &&)      = delete;
+  Span &operator=(const Span &) = delete;
+  Span &operator=(Span &&) = delete;
+
+  // TODO
+  // Sets an attribute on the Span. If the Span previously contained a mapping for
+  // the key, the old value is replaced.
+  //
+  // If an empty string is used as the value, the attribute will be silently
+  // dropped. Note: this behavior could change in the future.
+  // virtual void SetAttribute(nostd::string_view key, AttributeValue&& value) = 0;
+
+  // Adds an event to the Span.
+  virtual void AddEvent(nostd::string_view name) noexcept = 0;
+
+  // Adds an event to the Span, with a custom timestamp.
+  virtual void AddEvent(nostd::string_view name, core::SystemTimestamp timestamp) noexcept = 0;
+  virtual void AddEvent(nostd::string_view name, core::SteadyTimestamp timestamp) noexcept = 0;
+
+  // TODO
+  // Adds an event to the Span, with a custom timestamp, and attributes.
+  // virtual void AddEvent(nostd::string_view name, core::SteadyTimestamp
+  // timestamp, nostd::span<std::pair<nostd::string_view name, AttributeValue
+  // value>> attributes) noexcept = 0;
+
+  // Sets the status of the span. The default status is OK. Only the value of the last call will be
+  // recorded, and implementations are free to ignore previous calls.
+  virtual void SetStatus(CanonicalCode code, nostd::string_view description) noexcept = 0;
+
+  // Updates the name of the Span. If used, this will override the name provided
+  // during creation.
+  virtual void UpdateName(nostd::string_view name) noexcept = 0;
+
+  // Mark the end of the Span. Only the timing of the first end call for a given {@code Span} will
+  // be recorded, and implementations are free to ignore all further calls.
+  virtual void End() noexcept = 0;
+
+  // TODO
+  // virtual void End(EndSpanOptions&& opts) noexcept = 0;
+
+  // TODO
+  // SpanContext GetContext() const noexcept = 0;
```

Comment:
What's google's naming convention for accessors like this? I see [this section](https://google.github.io/styleguide/cppguide.html#Function_Names)
> Accessors and mutators (get and set functions) may be named like variables. These often correspond to actual member variables, but this is not required. For example, int count() and void set_count(int count)

I don't have a preference for using GetXyz or xyz style accessors; but if we're going to use this style, can we make the `tracer` method match.

Reply:
Renamed to `context` to follow the style guide. `GetContext` was meant to reflect the Java API but I guess sticking to C++ style is the right thing to do here.